coaster-nn/src/frameworks/native/mod.rs

```diff
@@ -175,8 +179,13 @@ where
         x: &SharedTensor<T>,
         result: &mut SharedTensor<T>,
         _workspace: &mut SharedTensor<u8>,
-        config: &Self::CC,
+        config: &dyn ConvolutionConfig<T>,
     ) -> Result<(), Error> {
+        let native_config = config
+            .as_any()
+            .downcast_ref::<helper::ConvolutionConfig>()
+            .unwrap();
```

Comment:
At what do we verify the correct type was passed in?

Reply:
> At what do we verify the correct type was passed in?

If the type is wrong, `downcast_ref()` will return `None` and `unwrap()` will panic.

But this was more of an experiment while I was trying to explore the viability of removing generic argument for `Layer`. As I mentioned in https://github.com/spearow/juice/pull/166#issuecomment-1191986691, I no longer think this is a good idea. 

To expand my reasoning a bit more, I now view the `LayerConfig` (in the current enum form) as the platform-independent representation of the net which can also be persisted across runs/platforms via capnp (together with weight values). The actual layers are an instantiation of the net for a specific platform, so genericizing it with the backend sort of makes sense.